pkg/platform/runtime/runtime.go

```diff
@@ -141,15 +141,32 @@ func (r *Runtime) validateCache() error {
 
 	// Check if local build script has changes that should be committed.
 	if r.cfg.GetBool(constants.OptinBuildscriptsConfig) {
-		if cachedScript, err := r.store.BuildScript(); err == nil {
+		cachedScript, err := r.store.BuildScript()
+		switch {
+		case err == nil:
 			script, err := buildscript.ScriptFromProject(r.target)
-			if err != nil {
+			if err != nil && !errs.Matches(err, buildscript.ErrBuildscriptNotExist) {
 				return errs.Wrap(err, "Unable to get local build script")
 			}
-			if !script.Equals(cachedScript) {
+			if script != nil && !script.Equals(cachedScript) {
 				return &NeedsCommitError{errs.New("Runtime changes should be committed")}
 			}
-		} else if err != store.ErrNoBuildScriptFile {
+
+		case err == store.ErrNoBuildScriptFile:
+			bp := model.NewBuildPlannerModel(r.auth)
+			expr, atTime, err := bp.GetBuildExpressionAndTime(commitID)
+			if err != nil {
+				return errs.Wrap(err, "Unable to get remote build expression and time")
+			}
+			script, err := buildscript.NewFromCommit(atTime, expr)
+			if err != nil {
+				return errs.Wrap(err, "Unable to get local build script")
+			}
+			if err := r.store.StoreBuildScript(script); err != nil {
+				return errs.Wrap(err, "Unable to store build script")
+			}
```

Comment:
This shouldn't be in `validateCache()`, it's implicitly creating the buildscript to satisfy some external use-case of this function that is not evident from within this function.

I think we'll want to just always store this buildscript once we have a runtime sourced. Same way we store the build expression I'm guessing.

This really isn't critical for v0.43, but I also don't want this to add more technical debt. I'd be ok taking this out entirely and filing a followup to address the opt-in route.

Reply:
I filed DX-2650 for this.

Reply:
Okay, I'll remove it.